src/content/tutorials/access-management/iam-roles-for-service-accounts/index.md

```diff
@@ -0,0 +1,123 @@
+description: This article describes how to use a new feature that allows binding of specific AWS IAM roles to a service account of a pod.
+weight: 60
+menu:
+  principal:
+    parent: tutorials-access-management
+    identifier: tutorials-access-management-irsa
+user_questions:
+ -  How can I use IAM roles for service accounts?
+last_review_date: 2024-12-03
+owner:
+  - https://github.com/orgs/giantswarm/teams/team-phoenix
+---
+
+*Basic knowledge about how service accounts work in Kubernetes is necessary for understanding what this page is about. If you could use a quick refresher, we strongly suggest taking a look at [this article](https://learnk8s.io/authentication-kubernetes).*
+
+You can associate an IAM role with a `Kubernetes` service account. This service account can then provide AWS permissions to the containers in any pod that uses that service account. With this feature, you no longer need to provide extended permissions to the Giant Swarm node IAM role so that pods on that node can call AWS APIs.
+
+Applications must sign their AWS API requests with AWS credentials. This feature provides a strategy for managing credentials for your applications, similar to the way that Amazon EC2 instance profiles provide credentials to Amazon EC2 instances. Instead of creating and distributing your AWS credentials to the containers or using the Amazon EC2 instance’s role, you can associate an IAM role with a `Kubernetes` service account. The applications in the pod’s containers can then use an AWS SDK or the AWS CLI to make API requests to authorized AWS services.
+
+The official documentation from AWS: [IAM roles for service accounts](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) `IRSA`.
+
+## IAM role permissions on `capa` controller
+
+The Cluster APi for AWs controller comes with an attached policy name `giantswarm-<INSTALLATION>-irsa-controller-policy` that contains permissions like:
+
+```json
+  {
+      "Effect": "Allow",
+      "Action": [
+        "iam:CreateOpenIDConnectProvider",
+        ...
+        "cloudfront:TagResource",
+        ...
+        "acm:RequestCertificate",
+        ...
+      ],
+      "Resource": "*"
+  },
+  {
+      "Effect": "Allow",
+      "Action": [
+        "s3:CreateBucket",
+        ...
+      ],
+      "Resource": "arn:aws:s3:::*-g8s-*"
+  }
+```
+
+This policy allows the controller to create a Cloudfront distribution, ACM certificate, S3 bucket and OpenID Connect provider to enable the usage of the IRSA feature.
+
+## Using IAM roles for service accounts
+
+### IAM role
+
+#### IAM Role Trusted Entity
+
+The **trusted entity** of a IAM role needs to contain a condition like the `sub` service account name of a JSON Web token (JWT). You need to ensure one of those conditions match against the `JWT`, which is used by one of your applications.
+
+This is a example `JWT` from `external-dns` which is verified against the `external-dns` IAM role.
```

Comment:
No need to go into technical details of JWTs at all. We need to give an example trust policy here that allows access by the Kubernetes service account.

Internal example:

```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Principal": {
                "Federated": "arn:aws:iam::242036376510:oidc-provider/irsa.golem.gaws.gigantic.io"
            },
            "Action": "sts:AssumeRoleWithWebIdentity",
            "Condition": {
                "StringEquals": {
                    "irsa.golem.gaws.gigantic.io:sub": "system:serviceaccount:mimir:mimir-alertmanager"
                }
            }
        }
    ]
}
```

Also, we should state whether customers should rely on the OIDC provider domain that comes with the WC or create their own.